libsignal-service/src/push_service.rs

```diff
@@ -574,35 +596,15 @@ pub trait PushService {
             .await?;
         assert!(!pre_key_response.devices.is_empty());
 
+        let identity = IdentityKey::decode(&pre_key_response.identity_key)?;
         let device = pre_key_response.devices.remove(0);
-        let mut bundle = PreKeyBundle::builder()
-            .identity_key(&PublicKey::decode_point(
-                &context,
-                &pre_key_response.identity_key,
-            )?)
-            .device_id(device.device_id)
-            .registration_id(device.registration_id);
-        if let Some(signed_pre_key) = device.signed_pre_key {
-            bundle = bundle.signed_pre_key(
-                signed_pre_key.key_id,
-                &PublicKey::decode_point(&context, &signed_pre_key.public_key)?,
-            );
-            bundle = bundle.signature(&signed_pre_key.signature);
-        }
-        if let Some(pre_key) = device.pre_key {
-            bundle = bundle.pre_key(
-                pre_key.key_id,
-                &PublicKey::decode_point(context, &pre_key.public_key)?,
-            );
-        }
-        Ok(bundle.build()?)
+        Ok(device.into_bundle(identity)?)
```

Comment:
@gferon, iirc I wrote this, could you check whether this seems alright wrt. `impl PreKeyResponseItem {  fn into_bundle(` a bit higher up?

Reply:
@gferon, can you verify whether this still checks out?

Reply:
Had a look at it and all of the fields, LGTM!